Production/plugins/SummaryProducer.cc

```diff
@@ -45,42 +48,34 @@ class SummaryProducer : public edm::EDAnalyzer {
             expressTuple = std::shared_ptr<ntuple::ExpressTuple>(
                     new ntuple::ExpressTuple("all_events", &edm::Service<TFileService>()->file(), false));
 
-        std::map<Channel, TriggerDescriptors> triggerDescriptors;
-        const auto& triggerSetup = cfg.getParameterSetVector("triggerSetup");
-        for(const auto& channelSetup : triggerSetup) {
-            const std::string channel_name = channelSetup.getParameter<std::string>("channel");
+        analysis::ConfigReader config_reader;
+
+        analysis::TriggerFileDescriptorCollection trigger_file_descriptors;
+        analysis::TriggerFileConfigEntryReader trigger_entry_reader(trigger_file_descriptors);
+        config_reader.AddEntryReader("PATTERN", trigger_entry_reader, true);
+        
+        analysis::SetupDescriptorCollection setup_descriptors;
+        analysis::SetupConfigEntryReader setup_entry_reader(setup_descriptors);
+        config_reader.AddEntryReader("SETUP", setup_entry_reader, false);
+	
+        const auto& triggerCfg = cfg.getParameter<std::string>("triggerCfg");
+        const std::string triggerCfg_full = edm::FileInPath(triggerCfg).fullPath();
+        config_reader.ReadConfig(triggerCfg_full);
```

Comment:
move this part of the code into a static method in TriggerTools and reuse in in both SummaryProducer and TriggerTools constructor.
E.g.: static TriggerFileDescriptorCollection ReadConfig(const std::string& cfg_path, triggers::SetupDescriptor& setup);